frontend/src/components/Forms/CoffeeChatsForm/CoffeeChatsDashboard.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { Card, Loader, Message, Button } from 'semantic-ui-react';
+import styles from './CoffeeChatsForm.module.css';
+import CoffeeChatAPI from '../../../API/CoffeeChatAPI';
+import { Emitters } from '../../../utils';
+
+const CoffeeChatsDashboard = (props: {
+  approvedChats: CoffeeChat[];
+  pendingChats: CoffeeChat[];
+  rejectedChats: CoffeeChat[];
+  isChatLoading: boolean;
+  setPendingChats: Dispatch<SetStateAction<CoffeeChat[]>>;
+}): JSX.Element => {
+  const { approvedChats, pendingChats, rejectedChats, isChatLoading, setPendingChats } = props;
+
+  const deleteCoffeeChatRequest = (coffeeChat: CoffeeChat) => {
+    CoffeeChatAPI.deleteCoffeeChat(coffeeChat.uuid)
+      .then(() => {
+        setPendingChats(pendingChats.filter((currChat) => currChat.uuid !== coffeeChat.uuid));
+        Emitters.generalSuccess.emit({
+          headerMsg: 'Coffee Chat Deleted!',
+          contentMsg: 'Your coffee chat was successfully deleted!'
+        });
+        Emitters.coffeeChatsUpdated.emit();
+      })
+      .catch((error) => {
+        Emitters.generalError.emit({
+          headerMsg: 'You are not allowed to delete this coffee chat!',
+          contentMsg: error
+        });
+      });
+  };
+
+  const CoffeeChatsDisplay = (props: { coffeeChatList: CoffeeChat[] }): JSX.Element => {
+    const { coffeeChatList } = props;
+
+    return (
+      <Card.Group>
+        {coffeeChatList.map((coffeeChat) => (
+          <Card key={coffeeChat.uuid}>
+            <Card.Content>
+              <Card.Header>Coffee Chat with {coffeeChat.otherMember.firstName}</Card.Header>
+
+              <div
+                style={{
+                  display: 'flex',
+                  justifyContent: 'space-between',
+                  alignItems: 'center',
+                  marginTop: '10px'
+                }}
+              >
+                <div>
+                  <Card.Meta>Category: {coffeeChat.slackLink}</Card.Meta>
+                  <Card.Meta>
+                    <b>
+                      <a href={coffeeChat.slackLink}>Image Link</a>
+                    </b>
+                  </Card.Meta>
+                </div>
+                <Card.Meta>
+                  {coffeeChat.status === 'pending' && (
+                    <div>
+                      <Button
+                        basic
+                        color="red"
+                        floated="right"
+                        size="small"
+                        onClick={() => {
+                          deleteCoffeeChatRequest(coffeeChat);
```

Comment:
Something about this delete button doesn't seem to be work (testing on the deploy preview). I tried hitting "Delete", and it appeared to delete, but it seems it was rejected by a 403:
```
{
    "error": "User with email axc2@cornell.edu does not have sufficient permissions to delete coffee chat."
}
```
(I disabled by admin access to test this) I think this is because we decided to set the delete permissions here to only allow admins, since I mentioned that permissions were going to be tough to manage.

I take back what I said! Since each member submits their own coffee chats, this isn't hard to manage, and we should allow people to delete a coffee chat on the API if they are the submitter.

Reply:
Also, accidentally caught a styling bug. If the category gets too long, the delete button appears to be pushed off the card. This usually shouldn't happen, but in case it does, I would expect the delete button to be pushed to the next line and for the category text to wraparound within the card.

![Screenshot 2024-10-05 at 9 27 38 PM](https://github.com/user-attachments/assets/f2181dd4-6852-4d6c-bfb9-e2f80b86aece)